packages/dds/tree/src/test/memory/tree.spec.ts

```diff
@@ -0,0 +1,146 @@
+			idCompressor: testIdCompressor,
+		}),
+		id,
+	);
+
+	const view = sharedTree.viewWith(new TreeViewConfiguration({ schema: RootNodeSchema }));
+
+	view.initialize(
+		new RootNodeSchema({
+			propertyOne: 128,
+			propertyTwo: {
+				itemOne: "",
+			},
+		}),
+	);
+
+	return view;
+}
+
+describe("SharedTree memory usage", () => {
+	// IMPORTANT: variables scoped to the test suite are a big problem for memory-profiling tests
+	// because they won't be out of scope when we garbage-collect between runs of the same test,
+	// and that will skew measurements. Tests should allocate all the memory they need using local
+	// variables scoped to the test function itself, so several iterations of a given test can
+	// measure from the same baseline (as much as possible).
+
+	beforeEach(async () => {
+		// CAREFUL: usually beforeEach/afterEach hooks are used to initialize or interact with variables
+		// whose scope is the encompasing test suite, but that's a problem for memory-profiling tests.
+		// See the comment at the top of the test suite for more details.
+	});
+
+	afterEach(() => {
+		// CAREFUL: usually beforeEach/afterEach hooks are used to initialize or interact with variables
+		// whose scope is the encompasing test suite, but that's a problem for memory-profiling tests.
+		// See the comment at the top of the test suite for more details.
+	});
+
+	benchmarkMemory(
+		new (class implements IMemoryTestObject {
+			public readonly title = "Create empty SharedTree";
+			public readonly minSampleCount = 500;
+
+			private sharedTree: TreeView<typeof RootNodeSchema> =
+				createLocalSharedTree("testSharedTree");
+
+			public async run(): Promise<void> {
+				this.sharedTree = createLocalSharedTree("testSharedTree");
+			}
+		})(),
+	);
+
+	const numbersOfEntriesForTests = [1000, 10_000, 100_000];
+
+	for (const x of numbersOfEntriesForTests) {
+		benchmarkMemory(
+			new (class implements IMemoryTestObject {
+				public readonly title = `Add ${x} integers to a local SharedTree`;
+				private sharedTree: TreeView<typeof RootNodeSchema> =
+					createLocalSharedTree("testSharedTree");
+
+				public async run(): Promise<void> {
+					for (let i = 0; i < x; i++) {
+						this.sharedTree.root.propertyOne = x;
+					}
+				}
+
+				public beforeIteration(): void {
+					this.sharedTree = createLocalSharedTree("testSharedTree");
+				}
+			})(),
+		);
+
+		benchmarkMemory(
+			new (class implements IMemoryTestObject {
+				public readonly title = `Add ${x} integers to a local SharedTree`;
+				private sharedTree: TreeView<typeof RootNodeSchema> =
+					createLocalSharedTree("testSharedTree");
+
+				public async run(): Promise<void> {
+					for (let i = 0; i < x; i++) {
+						this.sharedTree.root.propertyTwo.itemOne = i.toString().padStart(6, "0");
+					}
+				}
+
+				public beforeIteration(): void {
+					this.sharedTree = createLocalSharedTree("testSharedTree");
+				}
+			})(),
+		);
+
+		benchmarkMemory(
```

Comment:
This one is still planned to stay? Feels like the approach we discussed yesterday and wasn't quite right.

Reply:
Correct me if I'm wrong, but I thought we agreed that testing the removal of the _SharedTree content_ (not the ST itself) is valid?

Reply:
Yes, removing the content could be a valid test (that, as we discussed, we might see using a lot of memory because "removal" really just moves things around in the SharedTree's internal data structures). The test was doing something like trying to replace the tree though.

Reply:
Added `optional` field to the SharedTree.

Reply:
I think we don't need the extra "layer". `propertyOne` is already an optional number. We can just keep setting that property (and _only_ that property, don't update `propertyTwo` and `propertyThree`) to different integers inside the loop, and then `this.sharedTree.root.propertyOne = undefined`.

Reply:
I made changes to the last `benchMark()` test suite:
- Only sets the `propertyOne` to `x`.
- `this.sharedTree.root.child.propertyOne = undefined`